doc/source/index.rst

```diff
@@ -232,6 +232,22 @@ call::
 This means that the subjects of the messages you are interested in should start
 with "data_type1" though...
 
+Handling timezone-aware datetime objects
+----------------------------------------
+
+Timezone-aware datetime object were historically unsupported in posttroll, such as encoding or decoding them was
+leading to problems. Recent versions of posttroll were fixed to address the problem, however message sent with these
+versions are not backwards compatible. To ensure backwards compatibility, it is possible to configure posttroll to send
+message that drop the timezone information on encoding. This can be done with an environment variable::
+
+  POSTTROLL_MESSAGE_VERSION=v1.01
+
+or within python code::
+
+   >>> from posttroll import config
+   >>> with config.set(message_version="v1.01"):
+   ...
+
```

Comment:
Reading this it is possible to get the impression that also the datetimes within the message data could be affected. But as far as I've understood the code changes, it is actually only the header portion (the message creation time) that is affected and the times within the message data come from what ever created them.

Reply:
No, the metadata is also affected. That’s actually how we noticed the problem in the first place, as an older version of posttroll was refusing to decode the start_time item of the metadata and was leaving it as a string.

Reply:
Ok, got it. When receiving messages with tz aware datetimes, there's a problem decoding the messages. I didn't see this in my tests because the crash happened already with the message creation time.

So the processes earlier in the chain can't use tz aware datetimes within the message data unless the later parts have the fix. So the later parts will need the fix first, so they will already handle both aware and unaware datetimes. Nameserver only needs to understand the message creation time, as it doesn't need to actually parse the messages.

Did I get that right or did I miss something?

Reply:
That’s correct. Nameserver messages don’t have any encoded time in their body (it’s just text iirc).